From: Peter Xu <peterx@redhat.com>
Subject: fixup! mm: always initialise folio->_deferred_list
Date: Wed, 17 Apr 2024 17:18:34 -0400

Current mm-unstable will hit this when running test_hugetlb_memcg.  This
fixes the crash for me.

Link: https://lkml.kernel.org/r/20240417211836.2742593-2-peterx@redhat.com
Signed-off-by: Peter Xu <peterx@redhat.com>
Cc: David Hildenbrand <david@redhat.com>
Cc: "Matthew Wilcox (Oracle)" <willy@infradead.org>
Cc: Miaohe Lin <linmiaohe@huawei.com>
Cc: Muchun Song <muchun.song@linux.dev>
Cc: Oscar Salvador <osalvador@suse.de>
Cc: Vlastimil Babka <vbabka@suse.cz>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/memcontrol.c |    1 +
 1 file changed, 1 insertion(+)

--- a/mm/memcontrol.c~mm-always-initialise-folio-_deferred_list-fix
+++ a/mm/memcontrol.c
@@ -7401,6 +7401,7 @@ static void uncharge_folio(struct folio
 
 	VM_BUG_ON_FOLIO(folio_test_lru(folio), folio);
 	VM_BUG_ON_FOLIO(folio_order(folio) > 1 &&
+			!folio_test_hugetlb(folio) &&
 			!list_empty(&folio->_deferred_list), folio);
 
 	/*
_
